src/PHPS_fmt_plug.c

```diff
@@ -162,10 +163,12 @@ static int phps_valid(char *ciphertext, struct fmt_main *self)
 
 		if (!val)
 			return 0;
-		cp = strrchr(ciphertext, '$');
-		if (strstr(ciphertext, "$HEX$"))
-			wanted_len += 3; // salt is in hex.
-		return cp && strlen(cp) == wanted_len;
+		cp = &ciphertext[11 + MD5_HEX_SIZE];
```

Comment:
How do we know(?) we're not over-jumping beyond `ciphertext` length here?

Reply:
I hoped that `pDynamic_6->methods.valid` would catch shorter ciphertexts. I tried: it seems to reject them as expected.